app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeScreen.kt

```diff
@@ -0,0 +1,211 @@
+package agency.five.codebase.android.movieapp.ui.home
+
+import HomeMovieCategoryViewState
+import HomeMovieViewState
+import agency.five.codebase.android.movieapp.mock.MoviesMock.getMoviesList
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.component.MovieCard
+import agency.five.codebase.android.movieapp.ui.component.MovieCardViewState
+import agency.five.codebase.android.movieapp.ui.component.MovieCategoryLabel
+import agency.five.codebase.android.movieapp.ui.component.MovieCategoryLabelViewState
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapperImpl
+import agency.five.codebase.android.movieapp.ui.theme.CustomHeader
+import agency.five.codebase.android.movieapp.ui.theme.MovieAppTheme
+import agency.five.codebase.android.movieapp.ui.theme.Spacing
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+private val homeScreenMapper: HomeScreenMapper = HomeScreenMapperImpl()
+
+val popularCategoryViewState =
+    homeScreenMapper.toHomeMovieCategoryViewState(listOf(MovieCategory.POPULAR_STREAMING,
+        MovieCategory.POPULAR_ONTV,
+        MovieCategory.POPULAR_FORRENT,
+        MovieCategory.POPULAR_INTHEATRES), MovieCategory.POPULAR_STREAMING, getMoviesList())
+val nowPlayingCategoryViewState =
+    homeScreenMapper.toHomeMovieCategoryViewState(listOf(MovieCategory.NOWPLAYING_TV,
+        MovieCategory.NOWPLAYING_MOVIES), MovieCategory.NOWPLAYING_TV, getMoviesList())
+val upcomingCategoryViewState =
+    homeScreenMapper.toHomeMovieCategoryViewState(listOf(MovieCategory.UPCOMING_TODAY,
+        MovieCategory.UPCOMING_THISWEEK), MovieCategory.UPCOMING_TODAY, getMoviesList())
+
+@Composable
+fun HomeScreenRoute(
+    onNavigateToMovieDetails: (HomeMovieViewState) -> Unit,
+) {
+    var popularMovies by remember { mutableStateOf(popularCategoryViewState) }
+    var nowPlayingMovies by remember { mutableStateOf(nowPlayingCategoryViewState) }
+    var upcomingMovies by remember { mutableStateOf(upcomingCategoryViewState) }
+
+    HomeScreen(
+        popularMovies,
+        nowPlayingMovies,
+        upcomingMovies,
+        onNavigateToMovieDetails = onNavigateToMovieDetails,
+        onCategoryClick = {
+            when (it.itemId) {
+                0, 1, 2, 3 -> popularMovies =
```

Comment:
you can use ordinal from enum values instead of hardcoded numbers:

```kotlin
MovieCategory.POPULAR_STREAMING.ordinal,
MovieCategory.POPULAR_FOR_RENT.ordinal,
MovieCategory.POPULAR_ON_TV.ordinal,
MovieCategory.POPULAR_IN_THEATRES.ordinal,
->  {
   // ...
}
```